cmd/state-installer/cmd.go

```diff
@@ -355,12 +354,18 @@ func postInstallEvents(out output.Outputer, cfg *config.Instance, an analytics.D
 		return errs.Wrap(err, "Could not resolve installation path")
 	}
 
-	stateExe := appinfo.StateApp(installPath).Exec()
 	binPath, err := installation.BinPathFromInstallPath(installPath)
 	if err != nil {
 		return errs.Wrap(err, "Could not detect installation bin path")
 	}
 
+	stateInfo, err := installation.NewAppInfoInDir(installPath, installation.StateApp)
```

Comment:
This is slightly nitpicky, but I feel like this is still more complex than it needs to be. The appinfo architecture evolved organically and was never designed from the ground up. I feel like this PR is just perpetuating the organic design rather than revisiting it for the use-cases we have.

Looking at the data that AppInfo is returning (exec and name), I'm not seeing any reason for us to keep using this central struct based approach. It feels like we're trying to be too clever.

`installation.StateExec()` and `installation.StateName()` or something along those lines would service the exact same use-cases, without being needlessly clever. Dumb code that just works feels more desirable here because frankly there isn't much here for us to be smart with.

All of this I'm caveating as a nitpick though. Because technically all of this works fine, and I see no holes in the high level logic aside from that it feels like a roundabout way of facilitating the use-cases we have.